tests/generation/SettingsClassTest.py

```diff
@@ -0,0 +1,90 @@
+from tests.test_bears.AllKindsOfSettingsDependentBear import (
+    AllKindsOfSettingsDependentBear)
+from tests.test_bears.AllKindsOfSettingsDependentDecoratedBear import (
+    AllKindsOfSettingsDependentDecoratedBear)
+from tests.test_bears.AllKindsOfSettingsBaseBear import (
+    AllKindsOfSettingsBaseBear)
+from tests.test_bears.BearA import BearA
+
+
+class TestSettingsClass(unittest.TestCase):
+
+    def setUp(self):
+        self.printer = ConsolePrinter()
+        self.log_printer = None
+
+    def test_collect_bear_settings(self):
+        relevant_bears = {'test':
+                          {AllKindsOfSettingsDependentBear,
+                           AllKindsOfSettingsDependentDecoratedBear,
+                           AllKindsOfSettingsBaseBear,
+                           BearA}}
+
+        bear_settings_obj = collect_bear_settings(relevant_bears)
+
+        k = 0
+        for index, i in enumerate(bear_settings_obj):
+            if i.bear == AllKindsOfSettingsDependentBear:
+                k = index
+                break
+        obj = bear_settings_obj[k].non_optional_settings
+        self.assertCountEqual(obj.settings_bool, ['use_bear', 'use_bears'])
+        self.assertCountEqual(obj.settings_others, ['config',
+                                                    'max_line_lengths',
+                                                    'no_line', 'configs',
+                                                    'no_lines'])
+        obj = bear_settings_obj[k].optional_settings
+        self.assertCountEqual(obj.settings_bool, ['use_spaces', 'use_tabs',
+                                                  'chars'])
+        self.assertCountEqual(obj.settings_others, ['max_line_lengths',
+                                                    'no_chars',
+                                                    'dependency_results'])
+
+        for index, i in enumerate(bear_settings_obj):
+            if i.bear == AllKindsOfSettingsDependentDecoratedBear:
+                k = index
+                break
+        obj = bear_settings_obj[k].non_optional_settings
+        self.assertCountEqual(obj.settings_bool, ['use_bear', 'use_bears'])
+        self.assertCountEqual(obj.settings_others, ['configs',
+                                                    'config',
+                                                    'max_line_lengths',
+                                                    'no_line',
+                                                    'no_lines'])
+        obj = bear_settings_obj[k].optional_settings
+        self.assertCountEqual(obj.settings_bool, ['use_spaces', 'use_tabs',
+                                                  'chars'])
+        self.assertCountEqual(obj.settings_others, ['max_line_lengths',
+                                                    'no_chars',
+                                                    'dependency_results'])
+
+        for index, i in enumerate(bear_settings_obj):
```

Comment:
Usually, it's better to have separate functions for the tests (like [here](https://github.com/coala/coala/pull/3739/commits/7659c565a8a66c448ced642655b055b9a9e66acf#diff-8ab4f61ecc762c2ad5312562c454b2aaR212)), instead of batching multiple test scenarios in a single unittest testcase.

And if that seems like an overkill, just separate the test cases by adding a comment or two explaining what the test case is actually testing.